src/crate/client/sqlalchemy/tests/dict_test.py

```diff
@@ -90,19 +91,18 @@ def test_select_with_dict_column_where_clause_other_col(self):
         s = s.where(mytable.c.data['x'] == mytable.c.name)
         self.assertSQL(
             "SELECT mytable.name FROM mytable " +
-            "WHERE mytable.data['x'] = mytable.name",
+            "WHERE mytable.data[:data_1] = mytable.name",
             s
         )
 
     def test_update_with_dict_column(self):
         mytable = self.mytable
         stmt = mytable.update().\
             where(mytable.c.name == 'Arthur Dent').\
-            values({
-                "data['x']": "Trillian"
-            })
+            values({ mytable.c.data['x']: "Trillian" })
```

Comment:
This looks to be valid:
```python
from crate.client.sqlalchemy import types
from sqlalchemy.orm import Session
import sqlalchemy as sa

engine = sa.create_engine("crate://")
metadata_obj = sa.MetaData()

mytable = sa.Table(
    "characters",
    metadata_obj,
    sa.Column("id", sa.String, primary_key=True),
    sa.Column("details", types.Object),
    autoload_with=engine,
)

session = Session(bind=engine)
session.execute(
    mytable.update()
    .where(mytable.c.id == "1")
    .values({mytable.c.details["x"]: "Trillian"})
)
```

Resulting query as per `sys.jobs_log`: `UPDATE characters SET details['x'] = ? WHERE characters.id = ?`

Reply:
The previous test might have been wrong, using `values({"data['x']": "Trillian"})` throws an exception in a real-world example without any mocking: `TypeError: Object of type BindParameter is not JSON serializable`

Reply:
Oh. Thanks for spotting and investigating diligently.

Reply:
> throws an exception in a real-world example without any mocking

So, if that did not work before anyway, we will not have to announce anything as "breaking change", right? When looking at this again, I bet this was the spot which may have tripped me in the past already. Thanks again!

Reply:
I've added 2c6db67c to make the tests succeed.